contracts/lib/UniversalDeployer.Cairo

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.4.0b (utils/presets/UniversalDeployer.cairo)
+
+%lang starknet
+
+from starkware.starknet.common.syscalls import get_caller_address, deploy
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.hash import hash2
+from starkware.cairo.common.bool import FALSE, TRUE
+
+@event
+func ContractDeployed(
+    address: felt,
+    deployer: felt,
+    classHash: felt,
+    salt: felt
+) {
+}
+
+@external
+func deployContract{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr
+}(
+    classHash: felt,
+    salt: felt,
+    unique: felt,
+    calldata_len: felt,
+    calldata: felt*
+) -> (address: felt) {
+    
+    let (deployer) = get_caller_address();
+
+    tempvar prefix;
+    tempvar deploy_from_zero;
+    if (unique == TRUE) {
+        prefix = deployer;
+        deploy_from_zero = FALSE;
+    } else {
+        prefix = 'UniversalDeployerContract';
+        deploy_from_zero = TRUE;
+    }
+
+    let (_salt) = hash2{hash_ptr=pedersen_ptr}(prefix, salt);
```

Comment:
not sure if i get this correctly, but, does it mean we should use this same method to calculate the salt everywhere now? argent x, backend...

Reply:
The `UniversalDeployer` is still under discussion, this is just a mock contract to test that we can deploy at the correct address. This prefix is not needed for our use case so hopefully it will not be included in the final version.